src/needleman_wunch.jl

```diff
@@ -1,15 +1,145 @@
 # Labs 3 & 4
 
+
 """
-    nwscore(base1, base2)
+    nwscore(base1, base2, ; match= <int>, mismatch= <int>, gap= <int>)
+
+    returns an integer score depending on if the two bases are a match, mismatch, or gap
+
+    default scoring guidelines are as follows: match = 1, mismatch = -1, gap = -1 but can be changed
+    in function invocations as one wishes
+
+
+    Examples
+    ==============
+
+    julia> nwscore('G', 'C')
+    -1
+
+    julia> nwscore('C', 'A'; mismatch = -3)
+    -3
+
+    julia> nwscore('A', 'A')
+    1
+
+    julia> nwscore(Nothing, Nothing)
+    ArgumentError: "Score for two gaps is not defined"
+
+    julia> nwscore(Nothing, 'A'; gap = -2)
+    -2
+
 
-Your documentation here.
 """
-function nwscore(base1::Char, base2::Char)
-    
+function nwscore(base1::Char, base2::Char; match=1, mismatch=-1, gap=-1)
+    if base1 == base2
+        return match
+    else
+        return mismatch
+    end
 end
 
-nwscore(base::Char, ::Nothing) = -1
-nwscore(::Nothing, base::Char) = nwscore(base, nothing)
+nwscore(base::Char, ::Nothing; match=1, mismatch=-1, gap=-1) = gap
+nwscore(::Nothing, base::Char; match=1, mismatch=-1, gap=-1) = nwscore(base, nothing)
 
 nwscore(::Nothing, ::Nothing) = throw(ArgumentError("Score for two gaps is not defined"))
+
+#println(nwscore('G', 'C'))
+#println(nwscore('G', 'C'; match = 2, mismatch = -2))
+#println(nwscore('A', 'T'; mismatch=-2, match = 3, gap=-2))
+
+function nwsetupmatrix(seq1, seq2; gap = -1)
+    num_c = length(seq1) +1
+    num_r = length(seq2) +1
+    baseMatrix = zeros(Int, num_r, num_c)
+
+    if num_c > 1
```

Comment:
Interesting - this is good defensive coding. You never know what a user will throw at you! This would only be `false` if the sequence was empty. That should probably throw an error, though I think this will pass through and return a 1 column matrix filled with zeros.

This is beyond the scope of the lab, but another way to deal with this would be to, at the very top of the function, write something like

```
if isempty(seq1) || isempty(seq2) 
    throw(ArgumentError("Cannot setup matrix for sequence with length 0"))
end
```

or something.